demos/guide/quic-hq-interop-server.c

```diff
@@ -261,24 +261,34 @@ static SSL_CTX *create_ctx(const char *cert_path, const char *key_path)
  *   connections.
  *
  * Notes:
- * - This function assumes the use of IPv4 (`AF_INET`) and UDP (`SOCK_DGRAM`).
+ * - This function assumes UDP (`SOCK_DGRAM`).
+ * - This function accepts on both IPv4 and IPv6.
  * - The specified port is converted to network byte order using `htons`.
  */
 static BIO *create_socket(uint16_t port)
 {
     int fd = -1;
     BIO *sock = NULL;
     BIO_ADDR *addr = NULL;
-    struct in_addr ina;
+    int opt = 0;
+#ifdef _WIN32
+    struct in6_addr in6addr_any;
 
-    ina.s_addr = INADDR_ANY;
+    memset(&in6addr_any, 0, sizeof(in6addr_any));
+#endif
 
     /* Retrieve the file descriptor for a new UDP socket */
-    if ((fd = BIO_socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP, 0)) < 0) {
+    if ((fd = BIO_socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP, 0)) < 0) {
         fprintf(stderr, "cannot create socket");
         goto err;
     }
 
+    /* Disable IPV6_V6ONLY to accept both IPv4 and IPv6 */
+    if (setsockopt(fd, IPPROTO_IPV6, IPV6_V6ONLY, &opt, sizeof(opt)) < 0) {
```

Comment:
Note that this doesn't actually work on all platforms. At least OpenBSD doesn't support this.

Reply:
I think it's ok in this case since we just want this to work within the HQ QUIC interop runner (which it does currently)

Reply:
this should be fine as long as we don't build demos by default. the option seems to be available on windows and freebsd too. not sure about macos and netbsd.

Reply:
Should this be at least within `#ifdef IPV6_V6ONLY` ? So the interop server will build on systems where this is not defined.